Xwt.XamMac/Xwt.Mac/ListViewBackend.cs

```diff
@@ -140,14 +152,36 @@ void HandleDoubleClick (object sender, EventArgs e)
 				});
 			}
 		}
-		
+
+		public override NSTableColumn AddColumn (ListViewColumn col)
+		{
+			NSTableColumn tcol = base.AddColumn (col);
+			var widths = new List<nfloat> ();
+			if (Table.RowCount > 0)
+				widths.InsertRange (0, Enumerable.Repeat<nfloat> (-1f, (int)Table.RowCount));
+			ColumnRowWidths.Add (widths);
+			return tcol;
+		}
+
+		public override void RemoveColumn (ListViewColumn col, object handle)
+		{
+			var tcol = (NSTableColumn)handle;
+			var index = Columns.IndexOf (tcol);
+			ColumnRowWidths.RemoveAt (index);
+			base.RemoveColumn (col, handle);
+		}
+
 		public virtual void SetSource (IListDataSource source, IBackend sourceBackend)
 		{
 			this.source = source;
 
 			RowHeights = new List<nfloat> ();
 			for (int i = 0; i < source.RowCount; i++)
 				RowHeights.Add (-1);
+			foreach (var colWidths in ColumnRowWidths) {
+				colWidths.Clear ();
+				colWidths.AddRange (Enumerable.Repeat<nfloat> (-1, source.RowCount));
```

Comment:
Same as above, AddRange on `IEnumerable<T>` but not `ICollection<T>` does not do block copy.